core/src/apps/monero/xmr/chacha_poly.py

```
@@ -20,21 +20,20 @@ def _decrypt(
 
     cipher = ChaCha20Poly1305(key, iv)
     if associated_data:
         cipher.auth(associated_data)
     exp_tag, ciphertext = ciphertext[-16:], ciphertext[:-16]
     plaintext = cipher.decrypt(ciphertext)
-    tag = cipher.finish()
-    if not monero.ct_equals(tag, exp_tag):
+    try:
+        cipher.finish(exp_tag)
+    except RuntimeError:
         raise ValueError("tag invalid")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)